// @fluentui/react-virtualizer/lib-commonjs/components/VirtualizerScrollView/useVirtualizerScrollViewStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    virtualizerScrollViewClassNames: ()=>virtualizerScrollViewClassNames,
    useVirtualizerScrollViewStyles_unstable: ()=>useVirtualizerScrollViewStyles_unstable
});
const _useVirtualizerStyles = require("../Virtualizer/useVirtualizerStyles");
const _react = require("@griffel/react");
const virtualizerScrollViewClassName = 'fui-Virtualizer-Scroll-View';
const virtualizerScrollViewClassNames = {
    ..._useVirtualizerStyles.virtualizerClassNames,
    container: `${virtualizerScrollViewClassName}__container`
};
const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        mc9l5x: "f22iagw",
        a9b677: "fly5x3f",
        Bqenvij: "f1l02sjl",
        Eiaeu8: "f1115ve7"
    },
    vertical: {
        Beiy3e4: "f1vx9l62",
        Eiaeu8: "f1115ve7",
        Bmxbyg5: "f5zp4f"
    },
    horizontal: {
        Beiy3e4: "f1063pyq",
        B68tc82: "f1oy3dpc"
    },
    verticalReversed: {
        Beiy3e4: "f1gkdon0",
        Bmxbyg5: "f5zp4f"
    },
    horizontalReversed: {
        Beiy3e4: "f1oztnx0",
        B68tc82: "f1oy3dpc"
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".fly5x3f{width:100%;}",
        ".f1l02sjl{height:100%;}",
        ".f1115ve7{overflow-anchor:none;}",
        ".f1vx9l62{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}",
        ".f5zp4f{overflow-y:auto;}",
        ".f1063pyq{-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;}",
        ".f1oy3dpc{overflow-x:auto;}",
        ".f1gkdon0{-webkit-flex-direction:column-reverse;-ms-flex-direction:column-reverse;flex-direction:column-reverse;}",
        ".f1oztnx0{-webkit-flex-direction:row-reverse;-ms-flex-direction:row-reverse;flex-direction:row-reverse;}"
    ]
});
const useVirtualizerScrollViewStyles_unstable = (state)=>{
    const styles = useStyles();
    // For now - just return default style mods
    (0, _useVirtualizerStyles.useVirtualizerStyles_unstable)(state);
    const containerStyle = state.axis === 'horizontal' ? state.reversed ? styles.horizontalReversed : styles.horizontal : state.reversed ? styles.verticalReversed : styles.vertical;
    state.container.className = (0, _react.mergeClasses)(virtualizerScrollViewClassNames.container, styles.base, containerStyle, state.container.className);
    return state;
}; //# sourceMappingURL=useVirtualizerScrollViewStyles.js.map
