// @fluentui/react-virtualizer/lib-commonjs/components/VirtualizerScrollView/useVirtualizerScrollView.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useVirtualizerScrollView_unstable", {
    enumerable: true,
    get: ()=>useVirtualizerScrollView_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactUtilities = require("@fluentui/react-utilities");
const _useVirtualizer = require("../Virtualizer/useVirtualizer");
const _hooks = require("../../Hooks");
function useVirtualizerScrollView_unstable(props) {
    var _props_axis;
    const { virtualizerLength , bufferItems , bufferSize , scrollRef  } = (0, _hooks.useStaticVirtualizerMeasure)({
        defaultItemSize: props.itemSize,
        direction: (_props_axis = props.axis) !== null && _props_axis !== void 0 ? _props_axis : 'vertical'
    });
    const iScrollRef = (0, _reactUtilities.useMergedRefs)(_react.useRef(null), scrollRef);
    const virtualizerState = (0, _useVirtualizer.useVirtualizer_unstable)({
        ...props,
        virtualizerLength,
        bufferItems,
        bufferSize,
        scrollViewRef: iScrollRef
    });
    return {
        ...virtualizerState,
        components: {
            ...virtualizerState.components,
            container: 'div'
        },
        container: (0, _reactUtilities.resolveShorthand)(props.container, {
            required: true,
            defaultProps: {
                ref: iScrollRef
            }
        })
    };
} //# sourceMappingURL=useVirtualizerScrollView.js.map
