// @fluentui/react-text/lib-commonjs/components/presets/Subtitle1/Subtitle1.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    subtitle1ClassNames: ()=>subtitle1ClassNames,
    Subtitle1: ()=>Subtitle1
});
const _react = require("@griffel/react");
const _createPreset = require("../createPreset");
const subtitle1ClassNames = {
    root: 'fui-Subtitle1'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "f1pp30po",
        Bhrd7zp: "fl43uef",
        Bg96gwp: "f106mvju"
    }
}, {
    d: [
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".f1pp30po{font-size:var(--fontSizeBase500);}",
        ".fl43uef{font-weight:var(--fontWeightSemibold);}",
        ".f106mvju{line-height:var(--lineHeightBase500);}"
    ]
});
const Subtitle1 = /*#__PURE__*/ (0, _createPreset.createPreset)({
    useStyles,
    className: subtitle1ClassNames.root,
    displayName: 'Subtitle1'
}); //# sourceMappingURL=Subtitle1.js.map
