// @fluentui/react-tree/lib-commonjs/components/TreeItemLayout/useTreeItemLayoutStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    treeItemLayoutClassNames: ()=>treeItemLayoutClassNames,
    useTreeItemLayoutStyles_unstable: ()=>useTreeItemLayoutStyles_unstable
});
const _react = require("@griffel/react");
const _treeContext = require("../../contexts/treeContext");
const treeItemLayoutClassNames = {
    root: 'fui-TreeItemLayout',
    aside: 'fui-TreeItemLayout__aside',
    iconAfter: 'fui-TreeItemLayout__iconAfter',
    iconBefore: 'fui-TreeItemLayout__iconBefore'
};
/**
 * Styles for the root slot
 */ const useRootStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        mc9l5x: "f22iagw",
        Bt984gj: "f122n59",
        sshi5w: "f1nxs5xn",
        Bh6795r: "fqerorx",
        Bnnss6s: "f1neuvcm",
        xawz: "f1pztt34"
    },
    medium: {
        i8kkvl: "f1rjii52",
        Bahqtrf: "fk6fouc",
        Be2twd7: "fkhj508",
        Bhrd7zp: "figsok6",
        Bg96gwp: "f1i3iumi"
    },
    small: {
        i8kkvl: "f1ufnopg",
        sshi5w: "f1pha7fy",
        Bahqtrf: "fk6fouc",
        Be2twd7: "fy9rknc",
        Bhrd7zp: "figsok6",
        Bg96gwp: "fwrc4pm"
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
        ".f1nxs5xn{min-height:32px;}",
        ".fqerorx{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}",
        ".f1neuvcm{-webkit-flex-shrink:1;-ms-flex-negative:1;flex-shrink:1;}",
        ".f1pztt34{-webkit-flex-basis:0;-ms-flex-preferred-size:0;flex-basis:0;}",
        ".f1rjii52{-webkit-column-gap:var(--spacingHorizontalSNudge);column-gap:var(--spacingHorizontalSNudge);}",
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".fkhj508{font-size:var(--fontSizeBase300);}",
        ".figsok6{font-weight:var(--fontWeightRegular);}",
        ".f1i3iumi{line-height:var(--lineHeightBase300);}",
        ".f1ufnopg{-webkit-column-gap:var(--spacingHorizontalXS);column-gap:var(--spacingHorizontalXS);}",
        ".f1pha7fy{min-height:24px;}",
        ".fy9rknc{font-size:var(--fontSizeBase200);}",
        ".fwrc4pm{line-height:var(--lineHeightBase200);}"
    ]
});
/**
 * Styles for the before/after icon slot
 */ const useIconStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        mc9l5x: "f22iagw",
        Bt984gj: "f122n59",
        sj55zd: "fkfq4zb",
        Bg96gwp: "f106mvju",
        Be2twd7: "f1pp30po"
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
        ".fkfq4zb{color:var(--colorNeutralForeground2);}",
        ".f106mvju{line-height:var(--lineHeightBase500);}",
        ".f1pp30po{font-size:var(--fontSizeBase500);}"
    ]
});
/**
 * Styles for the action icon slot
 */ const useAsideStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        mc9l5x: "f22iagw",
        Bt984gj: "f122n59",
        Frg6f3: [
            "fcgxt0o",
            "f1ujusj6"
        ],
        z8tnut: "f1g0x7ka",
        z189sj: [
            "fw5db7e",
            "f1uw59to"
        ],
        Byoj8tv: "f1qch9an",
        uwmqm3: [
            "f1uw59to",
            "fw5db7e"
        ],
        i8kkvl: "f1ufnopg",
        Belr9w4: "f14sijuj"
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
        ".fcgxt0o{margin-left:auto;}",
        ".f1ujusj6{margin-right:auto;}",
        ".f1g0x7ka{padding-top:0;}",
        ".fw5db7e{padding-right:var(--spacingHorizontalM);}",
        ".f1uw59to{padding-left:var(--spacingHorizontalM);}",
        ".f1qch9an{padding-bottom:0;}",
        ".f1ufnopg{-webkit-column-gap:var(--spacingHorizontalXS);column-gap:var(--spacingHorizontalXS);}",
        ".f14sijuj{row-gap:var(--spacingHorizontalXS);}"
    ]
});
const useTreeItemLayoutStyles_unstable = (state)=>{
    const { iconAfter , iconBefore , aside , root  } = state;
    const rootStyles = useRootStyles();
    const iconStyles = useIconStyles();
    const asideStyles = useAsideStyles();
    const size = (0, _treeContext.useTreeContext_unstable)((ctx)=>ctx.size);
    root.className = (0, _react.mergeClasses)(treeItemLayoutClassNames.root, rootStyles.base, rootStyles[size], root.className);
    if (iconBefore) {
        iconBefore.className = (0, _react.mergeClasses)(treeItemLayoutClassNames.iconBefore, iconStyles.base, iconBefore.className);
    }
    if (iconAfter) {
        iconAfter.className = (0, _react.mergeClasses)(treeItemLayoutClassNames.iconAfter, iconStyles.base, iconAfter.className);
    }
    if (aside) {
        aside.className = (0, _react.mergeClasses)(treeItemLayoutClassNames.aside, asideStyles.base, aside.className);
    }
    return state;
}; //# sourceMappingURL=useTreeItemLayoutStyles.js.map
