// @fluentui/react-progress/lib-commonjs/components/ProgressField/ProgressField.js
/* eslint-disable deprecation/deprecation */ "use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    progressFieldClassNames: ()=>progressFieldClassNames,
    ProgressField: ()=>ProgressField
});
const _reactField = require("@fluentui/react-field");
const _progressBar = require("../../ProgressBar");
const progressFieldClassNames = /*#__PURE__*/ (0, _reactField.getDeprecatedFieldClassNames)(_progressBar.progressBarClassNames.root);
const ProgressField = /*#__PURE__*/ (0, _reactField.makeDeprecatedField)(_progressBar.ProgressBar, {
    displayName: 'ProgressField',
    mapProps: (props)=>({
            ...props,
            control: {
                ...props.control,
                validationState: props.validationState
            }
        })
}); //# sourceMappingURL=ProgressField.js.map
