// @fluentui/react-text/lib-commonjs/components/presets/Subtitle2/Subtitle2.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    subtitle2ClassNames: ()=>subtitle2ClassNames,
    Subtitle2: ()=>Subtitle2
});
const _react = require("@griffel/react");
const _createPreset = require("../createPreset");
const subtitle2ClassNames = {
    root: 'fui-Subtitle2'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "fod5ikn",
        Bhrd7zp: "fl43uef",
        Bg96gwp: "faaz57k"
    }
}, {
    d: [
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".fod5ikn{font-size:var(--fontSizeBase400);}",
        ".fl43uef{font-weight:var(--fontWeightSemibold);}",
        ".faaz57k{line-height:var(--lineHeightBase400);}"
    ]
});
const Subtitle2 = /*#__PURE__*/ (0, _createPreset.createPreset)({
    useStyles,
    className: subtitle2ClassNames.root,
    displayName: 'Subtitle2'
}); //# sourceMappingURL=Subtitle2.js.map
