// @fluentui/react-skeleton/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Skeleton: ()=>_skeleton.Skeleton,
    renderSkeleton_unstable: ()=>_skeleton.renderSkeleton_unstable,
    skeletonClassNames: ()=>_skeleton.skeletonClassNames,
    useSkeletonStyles_unstable: ()=>_skeleton.useSkeletonStyles_unstable,
    useSkeleton_unstable: ()=>_skeleton.useSkeleton_unstable,
    SkeletonItem: ()=>_skeletonItem.SkeletonItem,
    renderSkeletonItem_unstable: ()=>_skeletonItem.renderSkeletonItem_unstable,
    skeletonItemClassNames: ()=>_skeletonItem.skeletonItemClassNames,
    useSkeletonItemStyles_unstable: ()=>_skeletonItem.useSkeletonItemStyles_unstable,
    useSkeletonItem_unstable: ()=>_skeletonItem.useSkeletonItem_unstable,
    SkeletonContextProvider: ()=>_index.SkeletonContextProvider,
    useSkeletonContext: ()=>_index.useSkeletonContext
});
const _skeleton = require("./Skeleton");
const _skeletonItem = require("./SkeletonItem");
const _index = require("./contexts/index");
