// @fluentui/react-overflow/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Overflow: ()=>_overflow.Overflow,
    DATA_OVERFLOWING: ()=>_constants.DATA_OVERFLOWING,
    DATA_OVERFLOW_ITEM: ()=>_constants.DATA_OVERFLOW_ITEM,
    DATA_OVERFLOW_MENU: ()=>_constants.DATA_OVERFLOW_MENU,
    useIsOverflowGroupVisible: ()=>_useIsOverflowGroupVisible.useIsOverflowGroupVisible,
    useIsOverflowItemVisible: ()=>_useIsOverflowItemVisible.useIsOverflowItemVisible,
    useOverflowContainer: ()=>_useOverflowContainer.useOverflowContainer,
    useOverflowCount: ()=>_useOverflowCount.useOverflowCount,
    useOverflowItem: ()=>_useOverflowItem.useOverflowItem,
    useOverflowMenu: ()=>_useOverflowMenu.useOverflowMenu,
    useOverflowContext: ()=>_overflowContext.useOverflowContext,
    OverflowItem: ()=>_overflowItem.OverflowItem
});
const _overflow = require("./components/Overflow");
const _constants = require("./constants");
const _useIsOverflowGroupVisible = require("./useIsOverflowGroupVisible");
const _useIsOverflowItemVisible = require("./useIsOverflowItemVisible");
const _useOverflowContainer = require("./useOverflowContainer");
const _useOverflowCount = require("./useOverflowCount");
const _useOverflowItem = require("./useOverflowItem");
const _useOverflowMenu = require("./useOverflowMenu");
const _overflowContext = require("./overflowContext");
const _overflowItem = require("./components/OverflowItem/OverflowItem");
