// @fluentui/react-table/lib-commonjs/contexts/dataGridContext.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    DataGridContextProvider: ()=>DataGridContextProvider,
    useDataGridContext_unstable: ()=>useDataGridContext_unstable
});
const _reactContextSelector = require("@fluentui/react-context-selector");
const _hooks = require("../hooks");
const dataGridContext = /*#__PURE__*/ (0, _reactContextSelector.createContext)(undefined);
const dataGridContextDefaultValue = {
    ..._hooks.defaultTableState,
    subtleSelection: false,
    selectableRows: false,
    selectionAppearance: 'brand',
    focusMode: 'none'
};
const DataGridContextProvider = dataGridContext.Provider;
const useDataGridContext_unstable = (selector)=>(0, _reactContextSelector.useContextSelector)(dataGridContext, (ctx = dataGridContextDefaultValue)=>selector(ctx)); //# sourceMappingURL=dataGridContext.js.map
