// @fluentui/react-table/lib-commonjs/contexts/tableContext.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    tableContextDefaultValue: ()=>tableContextDefaultValue,
    TableContextProvider: ()=>TableContextProvider,
    useTableContext: ()=>useTableContext
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const tableContext = /*#__PURE__*/ _react.createContext(undefined);
const tableContextDefaultValue = {
    size: 'medium',
    noNativeElements: false,
    sortable: false
};
const TableContextProvider = tableContext.Provider;
var _React_useContext;
const useTableContext = ()=>(_React_useContext = _react.useContext(tableContext)) !== null && _React_useContext !== void 0 ? _React_useContext : tableContextDefaultValue; //# sourceMappingURL=tableContext.js.map
