// @fluentui/react-menu/lib-commonjs/components/MenuList/useMenuListStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    menuListClassNames: ()=>menuListClassNames,
    useMenuListStyles_unstable: ()=>useMenuListStyles_unstable
});
const _react = require("@griffel/react");
const menuListClassNames = {
    root: 'fui-MenuList'
};
const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        mc9l5x: "f22iagw",
        Beiy3e4: "f1vx9l62",
        i8kkvl: "f16mnhsx",
        Belr9w4: "fbi42co"
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f1vx9l62{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}",
        ".f16mnhsx{-webkit-column-gap:2px;column-gap:2px;}",
        ".fbi42co{row-gap:2px;}"
    ]
});
const useMenuListStyles_unstable = (state)=>{
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(menuListClassNames.root, styles.root, state.root.className);
    return state;
}; //# sourceMappingURL=useMenuListStyles.js.map
