// @fluentui/react-accordion/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Accordion: ()=>_accordion.Accordion,
    AccordionProvider: ()=>_accordion.AccordionProvider,
    accordionClassNames: ()=>_accordion.accordionClassNames,
    renderAccordion_unstable: ()=>_accordion.renderAccordion_unstable,
    useAccordionContext_unstable: ()=>_accordion.useAccordionContext_unstable,
    useAccordionContextValues_unstable: ()=>_accordion.useAccordionContextValues_unstable,
    useAccordionStyles_unstable: ()=>_accordion.useAccordionStyles_unstable,
    useAccordion_unstable: ()=>_accordion.useAccordion_unstable,
    AccordionItem: ()=>_accordionItem.AccordionItem,
    AccordionItemProvider: ()=>_accordionItem.AccordionItemProvider,
    accordionItemClassNames: ()=>_accordionItem.accordionItemClassNames,
    renderAccordionItem_unstable: ()=>_accordionItem.renderAccordionItem_unstable,
    useAccordionItemContextValues_unstable: ()=>_accordionItem.useAccordionItemContextValues_unstable,
    useAccordionItemContext_unstable: ()=>_accordionItem.useAccordionItemContext_unstable,
    useAccordionItemStyles_unstable: ()=>_accordionItem.useAccordionItemStyles_unstable,
    useAccordionItem_unstable: ()=>_accordionItem.useAccordionItem_unstable,
    AccordionHeader: ()=>_accordionHeader.AccordionHeader,
    accordionHeaderClassNames: ()=>_accordionHeader.accordionHeaderClassNames,
    renderAccordionHeader_unstable: ()=>_accordionHeader.renderAccordionHeader_unstable,
    useAccordionHeaderContextValues_unstable: ()=>_accordionHeader.useAccordionHeaderContextValues_unstable,
    useAccordionHeaderStyles_unstable: ()=>_accordionHeader.useAccordionHeaderStyles_unstable,
    useAccordionHeader_unstable: ()=>_accordionHeader.useAccordionHeader_unstable,
    AccordionPanel: ()=>_accordionPanel.AccordionPanel,
    accordionPanelClassNames: ()=>_accordionPanel.accordionPanelClassNames,
    renderAccordionPanel_unstable: ()=>_accordionPanel.renderAccordionPanel_unstable,
    useAccordionPanelStyles_unstable: ()=>_accordionPanel.useAccordionPanelStyles_unstable,
    useAccordionPanel_unstable: ()=>_accordionPanel.useAccordionPanel_unstable
});
const _accordion = require("./Accordion");
const _accordionItem = require("./AccordionItem");
const _accordionHeader = require("./AccordionHeader");
const _accordionPanel = require("./AccordionPanel");
