// @fluentui/react-text/lib-commonjs/components/presets/Body1Strong/Body1Strong.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    body1StrongClassNames: ()=>body1StrongClassNames,
    Body1Strong: ()=>Body1Strong
});
const _react = require("@griffel/react");
const _createPreset = require("../createPreset");
const body1StrongClassNames = {
    root: 'fui-Body1Strong'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "fkhj508",
        Bhrd7zp: "fl43uef",
        Bg96gwp: "f1i3iumi"
    }
}, {
    d: [
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".fkhj508{font-size:var(--fontSizeBase300);}",
        ".fl43uef{font-weight:var(--fontWeightSemibold);}",
        ".f1i3iumi{line-height:var(--lineHeightBase300);}"
    ]
});
const Body1Strong = /*#__PURE__*/ (0, _createPreset.createPreset)({
    useStyles,
    className: body1StrongClassNames.root,
    displayName: 'Body1Strong'
}); //# sourceMappingURL=Body1Strong.js.map
