// @fluentui/react-tree/lib-commonjs/components/TreeItemPersonaLayout/useTreeItemPersonaLayoutStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    treeItemPersonaLayoutClassNames: ()=>treeItemPersonaLayoutClassNames,
    useTreeItemPersonaLayoutStyles_unstable: ()=>useTreeItemPersonaLayoutStyles_unstable
});
const _react = require("@griffel/react");
const treeItemPersonaLayoutClassNames = {
    root: 'fui-TreeItemPersonaLayout',
    media: 'fui-TreeItemPersonaLayout__media',
    content: 'fui-TreeItemPersonaLayout__content',
    description: 'fui-TreeItemPersonaLayout__description',
    aside: 'fui-TreeItemPersonaLayout__aside',
    main: 'fui-TreeItemPersonaLayout__main'
};
/**
 * Styles for the root slot
 */ const useRootStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        a9b677: "fly5x3f",
        mc9l5x: "f22iagw",
        Bt984gj: "f122n59"
    }
}, {
    d: [
        ".fly5x3f{width:100%;}",
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}"
    ]
});
/**
 * Styles for the expand icon slot
 */ const useMediaStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        mc9l5x: "f22iagw",
        Bt984gj: "f122n59",
        a9b677: "f1szoe96",
        Bqenvij: "f1d2rq10",
        z8tnut: "f1g0x7ka",
        z189sj: [
            "f7x41pl",
            "fruq291"
        ],
        Byoj8tv: "f1qch9an",
        uwmqm3: [
            "fgiv446",
            "ffczdla"
        ]
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
        ".f1szoe96{width:32px;}",
        ".f1d2rq10{height:32px;}",
        ".f1g0x7ka{padding-top:0;}",
        ".f7x41pl{padding-right:var(--spacingHorizontalXS);}",
        ".fruq291{padding-left:var(--spacingHorizontalXS);}",
        ".f1qch9an{padding-bottom:0;}",
        ".fgiv446{padding-left:var(--spacingHorizontalXXS);}",
        ".ffczdla{padding-right:var(--spacingHorizontalXXS);}"
    ]
});
const useContentStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        a9b677: "fly5x3f",
        mc9l5x: "f22iagw",
        Beiy3e4: "f1vx9l62",
        z8tnut: "f1ngh7ph",
        z189sj: [
            "f7x41pl",
            "fruq291"
        ],
        Byoj8tv: "f5o476b",
        uwmqm3: [
            "f1f5gg8d",
            "f1vdfbxk"
        ],
        i8kkvl: "f1fps8zv",
        Belr9w4: "fws515f"
    }
}, {
    d: [
        ".fly5x3f{width:100%;}",
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f1vx9l62{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}",
        ".f1ngh7ph{padding-top:var(--spacingVerticalMNudge);}",
        ".f7x41pl{padding-right:var(--spacingHorizontalXS);}",
        ".fruq291{padding-left:var(--spacingHorizontalXS);}",
        ".f5o476b{padding-bottom:var(--spacingVerticalMNudge);}",
        ".f1f5gg8d{padding-left:var(--spacingHorizontalS);}",
        ".f1vdfbxk{padding-right:var(--spacingHorizontalS);}",
        ".f1fps8zv{-webkit-column-gap:var(--spacingVerticalNone);column-gap:var(--spacingVerticalNone);}",
        ".fws515f{row-gap:var(--spacingHorizontalNone);}"
    ]
});
const useMainStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "fkhj508",
        Bhrd7zp: "figsok6",
        Bg96gwp: "f1i3iumi"
    }
}, {
    d: [
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".fkhj508{font-size:var(--fontSizeBase300);}",
        ".figsok6{font-weight:var(--fontWeightRegular);}",
        ".f1i3iumi{line-height:var(--lineHeightBase300);}"
    ]
});
const useDescriptionStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "fy9rknc",
        Bhrd7zp: "figsok6",
        Bg96gwp: "fwrc4pm"
    }
}, {
    d: [
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".fy9rknc{font-size:var(--fontSizeBase200);}",
        ".figsok6{font-weight:var(--fontWeightRegular);}",
        ".fwrc4pm{line-height:var(--lineHeightBase200);}"
    ]
});
const useAsideStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        mc9l5x: "f22iagw",
        Beiy3e4: "f1vx9l62",
        Huce71: "fz5stix",
        z189sj: [
            "fw5db7e",
            "f1uw59to"
        ],
        Bahqtrf: "fk6fouc",
        Be2twd7: "fy9rknc",
        Bhrd7zp: "figsok6",
        Bg96gwp: "fwrc4pm"
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f1vx9l62{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}",
        ".fz5stix{white-space:nowrap;}",
        ".fw5db7e{padding-right:var(--spacingHorizontalM);}",
        ".f1uw59to{padding-left:var(--spacingHorizontalM);}",
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".fy9rknc{font-size:var(--fontSizeBase200);}",
        ".figsok6{font-weight:var(--fontWeightRegular);}",
        ".fwrc4pm{line-height:var(--lineHeightBase200);}"
    ]
});
const useTreeItemPersonaLayoutStyles_unstable = (state)=>{
    const rootStyles = useRootStyles();
    const mediaStyles = useMediaStyles();
    const contentStyles = useContentStyles();
    const mainStyles = useMainStyles();
    const descriptionStyles = useDescriptionStyles();
    const asideStyles = useAsideStyles();
    state.root.className = (0, _react.mergeClasses)(treeItemPersonaLayoutClassNames.root, rootStyles.base, state.root.className);
    state.media.className = (0, _react.mergeClasses)(treeItemPersonaLayoutClassNames.media, mediaStyles.base, state.media.className);
    if (state.content) {
        state.content.className = (0, _react.mergeClasses)(treeItemPersonaLayoutClassNames.content, contentStyles.base, state.content.className);
    }
    if (state.main) {
        state.main.className = (0, _react.mergeClasses)(treeItemPersonaLayoutClassNames.main, mainStyles.base, state.main.className);
    }
    if (state.description) {
        state.description.className = (0, _react.mergeClasses)(treeItemPersonaLayoutClassNames.description, descriptionStyles.base, state.description.className);
    }
    if (state.aside) {
        state.aside.className = (0, _react.mergeClasses)(treeItemPersonaLayoutClassNames.aside, asideStyles.base, state.aside.className);
    }
    return state;
}; //# sourceMappingURL=useTreeItemPersonaLayoutStyles.js.map
