// @fluentui/react-text/lib-commonjs/components/presets/Body1/Body1.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    body1ClassNames: ()=>body1ClassNames,
    Body1: ()=>Body1
});
const _react = require("@griffel/react");
const _createPreset = require("../createPreset");
const body1ClassNames = {
    root: 'fui-Body1'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "fkhj508",
        Bhrd7zp: "figsok6",
        Bg96gwp: "f1i3iumi"
    }
}, {
    d: [
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".fkhj508{font-size:var(--fontSizeBase300);}",
        ".figsok6{font-weight:var(--fontWeightRegular);}",
        ".f1i3iumi{line-height:var(--lineHeightBase300);}"
    ]
});
const Body1 = /*#__PURE__*/ (0, _createPreset.createPreset)({
    useStyles,
    className: body1ClassNames.root,
    displayName: 'Body1'
}); //# sourceMappingURL=Body1.js.map
