// @fluentui/tokens/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    teamsDarkTheme: ()=>_index.teamsDarkTheme,
    teamsHighContrastTheme: ()=>_index.teamsHighContrastTheme,
    teamsLightTheme: ()=>_index.teamsLightTheme,
    webDarkTheme: ()=>_index.webDarkTheme,
    webLightTheme: ()=>_index.webLightTheme,
    createDarkTheme: ()=>_index1.createDarkTheme,
    createHighContrastTheme: ()=>_index1.createHighContrastTheme,
    createLightTheme: ()=>_index1.createLightTheme,
    createTeamsDarkTheme: ()=>_index1.createTeamsDarkTheme,
    themeToTokensObject: ()=>_themeToTokensObject.themeToTokensObject,
    tokens: ()=>_tokens.tokens,
    typographyStyles: ()=>_index2.typographyStyles
});
const _index = require("./themes/index");
const _index1 = require("./utils/index");
const _themeToTokensObject = require("./themeToTokensObject");
const _tokens = require("./tokens");
const _index2 = require("./global/index");
