// @fluentui/react-toolbar/lib-commonjs/components/ToolbarGroup/useToolbarGroupStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    toolbarGroupClassNames: ()=>toolbarGroupClassNames,
    useToolbarGroupStyles_unstable: ()=>useToolbarGroupStyles_unstable
});
const _react = require("@griffel/react");
const toolbarGroupClassNames = {
    root: 'fui-ToolbarGroup'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        mc9l5x: "f22iagw",
        i8kkvl: "f4px1ci",
        Belr9w4: "fn67r4l"
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f4px1ci{-webkit-column-gap:8px;column-gap:8px;}",
        ".fn67r4l{row-gap:8px;}"
    ]
});
const useToolbarGroupStyles_unstable = (state)=>{
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(toolbarGroupClassNames.root, styles.root, state.root.className);
    return state;
}; //# sourceMappingURL=useToolbarGroupStyles.js.map
