// @fluentui/react-button/lib-commonjs/utils/useToggleState.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useToggleState", {
    enumerable: true,
    get: ()=>useToggleState
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactUtilities = require("@fluentui/react-utilities");
function useToggleState(props, state) {
    const { checked , defaultChecked , disabled , disabledFocusable  } = props;
    const { onClick , role  } = state.root;
    const [checkedValue, setCheckedValue] = (0, _reactUtilities.useControllableState)({
        state: checked,
        defaultState: defaultChecked,
        initialState: false
    });
    const isCheckboxTypeRole = role === 'menuitemcheckbox' || role === 'checkbox';
    const onToggleClick = _react.useCallback((ev)=>{
        if (!disabled && !disabledFocusable) {
            if (ev.defaultPrevented) {
                return;
            }
            setCheckedValue(!checkedValue);
        }
    }, [
        checkedValue,
        disabled,
        disabledFocusable,
        setCheckedValue
    ]);
    return {
        ...state,
        checked: checkedValue,
        root: {
            ...state.root,
            [isCheckboxTypeRole ? 'aria-checked' : 'aria-pressed']: checkedValue,
            onClick: (0, _reactUtilities.useEventCallback)((0, _reactUtilities.mergeCallbacks)(onClick, onToggleClick))
        }
    };
} //# sourceMappingURL=useToggleState.js.map
