// @fluentui/react-tree/lib-commonjs/contexts/treeContext.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    TreeContext: ()=>TreeContext,
    TreeProvider: ()=>TreeProvider,
    useTreeContext_unstable: ()=>useTreeContext_unstable
});
const _reactContextSelector = require("@fluentui/react-context-selector");
const _immutableSet = require("../utils/ImmutableSet");
const defaultContextValue = {
    level: 0,
    openItems: _immutableSet.emptyImmutableSet,
    requestOpenChange: noop,
    requestNavigation: noop,
    appearance: 'subtle',
    size: 'medium'
};
function noop() {
/* noop */ }
const TreeContext = /*#__PURE__*/ (0, _reactContextSelector.createContext)(undefined);
const { Provider: TreeProvider  } = TreeContext;
const useTreeContext_unstable = (selector)=>(0, _reactContextSelector.useContextSelector)(TreeContext, (ctx = defaultContextValue)=>selector(ctx)); //# sourceMappingURL=treeContext.js.map
