// @fluentui/react-tree/lib-commonjs/components/Tree/useTreeStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    treeClassNames: ()=>treeClassNames,
    useTreeStyles_unstable: ()=>useTreeStyles_unstable
});
const _react = require("@griffel/react");
const treeClassNames = {
    root: 'fui-Tree'
};
const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        mc9l5x: "f22iagw",
        Beiy3e4: "f1vx9l62",
        Belr9w4: "f1j0q4x9"
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f1vx9l62{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}",
        ".f1j0q4x9{row-gap:var(--spacingVerticalXXS);}"
    ]
});
const useTreeStyles_unstable = (state)=>{
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(treeClassNames.root, styles.root, state.root.className);
    return state;
}; //# sourceMappingURL=useTreeStyles.js.map
