// @fluentui/react-combobox/lib-commonjs/components/OptionGroup/useOptionGroup.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useOptionGroup_unstable", {
    enumerable: true,
    get: ()=>useOptionGroup_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactUtilities = require("@fluentui/react-utilities");
const useOptionGroup_unstable = (props, ref)=>{
    const labelId = (0, _reactUtilities.useId)('group-label');
    const { label  } = props;
    return {
        components: {
            root: 'div',
            label: 'span'
        },
        root: (0, _reactUtilities.getNativeElementProps)('div', {
            ref,
            role: 'group',
            'aria-labelledby': label ? labelId : undefined,
            ...props
        }),
        label: (0, _reactUtilities.resolveShorthand)(label, {
            defaultProps: {
                id: labelId,
                role: 'presentation'
            }
        })
    };
}; //# sourceMappingURL=useOptionGroup.js.map
