// @fluentui/react-badge/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Badge: ()=>_badge.Badge,
    badgeClassNames: ()=>_badge.badgeClassNames,
    renderBadge_unstable: ()=>_badge.renderBadge_unstable,
    useBadgeStyles_unstable: ()=>_badge.useBadgeStyles_unstable,
    useBadge_unstable: ()=>_badge.useBadge_unstable,
    PresenceBadge: ()=>_presenceBadge.PresenceBadge,
    presenceBadgeClassNames: ()=>_presenceBadge.presenceBadgeClassNames,
    usePresenceBadgeStyles_unstable: ()=>_presenceBadge.usePresenceBadgeStyles_unstable,
    usePresenceBadge_unstable: ()=>_presenceBadge.usePresenceBadge_unstable,
    CounterBadge: ()=>_counterBadge.CounterBadge,
    counterBadgeClassNames: ()=>_counterBadge.counterBadgeClassNames,
    useCounterBadgeStyles_unstable: ()=>_counterBadge.useCounterBadgeStyles_unstable,
    useCounterBadge_unstable: ()=>_counterBadge.useCounterBadge_unstable
});
const _badge = require("./Badge");
const _presenceBadge = require("./PresenceBadge");
const _counterBadge = require("./CounterBadge");
