// @fluentui/react-text/lib-commonjs/components/presets/Title1/Title1.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    title1ClassNames: ()=>title1ClassNames,
    Title1: ()=>Title1
});
const _react = require("@griffel/react");
const _createPreset = require("../createPreset");
const title1ClassNames = {
    root: 'fui-Title1'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "fccw675",
        Bhrd7zp: "fl43uef",
        Bg96gwp: "f1ebx5kk"
    }
}, {
    d: [
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".fccw675{font-size:var(--fontSizeHero800);}",
        ".fl43uef{font-weight:var(--fontWeightSemibold);}",
        ".f1ebx5kk{line-height:var(--lineHeightHero800);}"
    ]
});
const Title1 = /*#__PURE__*/ (0, _createPreset.createPreset)({
    useStyles,
    className: title1ClassNames.root,
    displayName: 'Title1'
}); //# sourceMappingURL=Title1.js.map
