// @fluentui/react-accordion/lib-commonjs/components/AccordionHeader/AccordionHeaderContext.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    AccordionHeaderContext: ()=>AccordionHeaderContext,
    useAccordionHeaderContext: ()=>useAccordionHeaderContext
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const AccordionHeaderContext = /*#__PURE__*/ _react.createContext(undefined);
const accordionHeaderContextDefaultValue = {
    open: false,
    disabled: false,
    size: 'medium',
    expandIconPosition: 'start'
};
var _React_useContext;
const useAccordionHeaderContext = ()=>(_React_useContext = _react.useContext(AccordionHeaderContext)) !== null && _React_useContext !== void 0 ? _React_useContext : accordionHeaderContextDefaultValue; //# sourceMappingURL=AccordionHeaderContext.js.map
