// @fluentui/react-accordion/lib-commonjs/components/AccordionItem/AccordionItemContext.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    AccordionItemContext: ()=>AccordionItemContext,
    AccordionItemProvider: ()=>AccordionItemProvider,
    useAccordionItemContext_unstable: ()=>useAccordionItemContext_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const AccordionItemContext = /*#__PURE__*/ _react.createContext(undefined);
const accordionItemContextDefaultValue = {
    onHeaderClick () {
    /** */ },
    open: false,
    disabled: false
};
const AccordionItemProvider = AccordionItemContext.Provider;
var _React_useContext;
const useAccordionItemContext_unstable = ()=>(_React_useContext = _react.useContext(AccordionItemContext)) !== null && _React_useContext !== void 0 ? _React_useContext : accordionItemContextDefaultValue; //# sourceMappingURL=AccordionItemContext.js.map
