// @fluentui/react-card/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Card: ()=>_card.Card,
    cardClassNames: ()=>_card.cardClassNames,
    cardCSSVars: ()=>_card.cardCSSVars,
    renderCard_unstable: ()=>_card.renderCard_unstable,
    useCardStyles_unstable: ()=>_card.useCardStyles_unstable,
    useCard_unstable: ()=>_card.useCard_unstable,
    CardFooter: ()=>_cardFooter.CardFooter,
    cardFooterClassNames: ()=>_cardFooter.cardFooterClassNames,
    renderCardFooter_unstable: ()=>_cardFooter.renderCardFooter_unstable,
    useCardFooterStyles_unstable: ()=>_cardFooter.useCardFooterStyles_unstable,
    useCardFooter_unstable: ()=>_cardFooter.useCardFooter_unstable,
    CardHeader: ()=>_cardHeader.CardHeader,
    cardHeaderClassNames: ()=>_cardHeader.cardHeaderClassNames,
    cardHeaderCSSVars: ()=>_cardHeader.cardHeaderCSSVars,
    renderCardHeader_unstable: ()=>_cardHeader.renderCardHeader_unstable,
    useCardHeaderStyles_unstable: ()=>_cardHeader.useCardHeaderStyles_unstable,
    useCardHeader_unstable: ()=>_cardHeader.useCardHeader_unstable,
    CardPreview: ()=>_cardPreview.CardPreview,
    cardPreviewClassNames: ()=>_cardPreview.cardPreviewClassNames,
    renderCardPreview_unstable: ()=>_cardPreview.renderCardPreview_unstable,
    useCardPreviewStyles_unstable: ()=>_cardPreview.useCardPreviewStyles_unstable,
    useCardPreview_unstable: ()=>_cardPreview.useCardPreview_unstable,
    CardProvider: ()=>_card.CardProvider,
    useCardContext_unstable: ()=>_card.useCardContext_unstable
});
const _card = require("./Card");
const _cardFooter = require("./CardFooter");
const _cardHeader = require("./CardHeader");
const _cardPreview = require("./CardPreview");
