// @fluentui/example-data/lib-amd/facepile.js
define(["require", "exports", "./testImages"], function (require, exports, testImages_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.facepilePersonas = void 0;
    /**
     * For use in this package only.
     * Mirror of PersonaInitialsColor avoid a circular dependency.
     * If the real enum changes and this one starts causing compiler errors, update it.
     * @internal
     */
    var PersonaInitialsColor;
    (function (PersonaInitialsColor) {
        PersonaInitialsColor[PersonaInitialsColor["lightBlue"] = 0] = "lightBlue";
        PersonaInitialsColor[PersonaInitialsColor["blue"] = 1] = "blue";
        PersonaInitialsColor[PersonaInitialsColor["darkBlue"] = 2] = "darkBlue";
        PersonaInitialsColor[PersonaInitialsColor["teal"] = 3] = "teal";
        PersonaInitialsColor[PersonaInitialsColor["lightGreen"] = 4] = "lightGreen";
        PersonaInitialsColor[PersonaInitialsColor["green"] = 5] = "green";
        PersonaInitialsColor[PersonaInitialsColor["darkGreen"] = 6] = "darkGreen";
        PersonaInitialsColor[PersonaInitialsColor["lightPink"] = 7] = "lightPink";
        PersonaInitialsColor[PersonaInitialsColor["pink"] = 8] = "pink";
        PersonaInitialsColor[PersonaInitialsColor["magenta"] = 9] = "magenta";
        PersonaInitialsColor[PersonaInitialsColor["purple"] = 10] = "purple";
        PersonaInitialsColor[PersonaInitialsColor["black"] = 11] = "black";
        PersonaInitialsColor[PersonaInitialsColor["orange"] = 12] = "orange";
        PersonaInitialsColor[PersonaInitialsColor["red"] = 13] = "red";
        PersonaInitialsColor[PersonaInitialsColor["darkRed"] = 14] = "darkRed";
        PersonaInitialsColor[PersonaInitialsColor["transparent"] = 15] = "transparent";
        PersonaInitialsColor[PersonaInitialsColor["violet"] = 16] = "violet";
        PersonaInitialsColor[PersonaInitialsColor["lightRed"] = 17] = "lightRed";
        PersonaInitialsColor[PersonaInitialsColor["gold"] = 18] = "gold";
        PersonaInitialsColor[PersonaInitialsColor["burgundy"] = 19] = "burgundy";
        PersonaInitialsColor[PersonaInitialsColor["warmGray"] = 20] = "warmGray";
        PersonaInitialsColor[PersonaInitialsColor["coolGray"] = 21] = "coolGray";
        PersonaInitialsColor[PersonaInitialsColor["gray"] = 22] = "gray";
        PersonaInitialsColor[PersonaInitialsColor["cyan"] = 23] = "cyan";
        PersonaInitialsColor[PersonaInitialsColor["rust"] = 24] = "rust";
    })(PersonaInitialsColor || (PersonaInitialsColor = {}));
    /** @internal */
    exports.facepilePersonas = [
        {
            imageUrl: testImages_1.TestImages.personaFemale,
            personaName: 'Annie Lindqvist',
            data: '50%',
            onClick: function (ev, persona) {
                return alert('You clicked on ' + persona.personaName + '. Extra data: ' + persona.data);
            },
        },
        {
            imageUrl: testImages_1.TestImages.personaFemale,
            personaName: 'Aaron Reid',
            data: '$1,000',
        },
        {
            personaName: 'Alex Lundberg',
            data: '75%',
            onClick: function (ev, persona) {
                return alert('You clicked on ' + persona.personaName + '. Extra data: ' + persona.data);
            },
        },
        {
            personaName: 'Roko Kolar',
            data: '4 hrs',
        },
        {
            imageInitials: 'CB',
            personaName: 'Christian Bergqvist',
            initialsColor: PersonaInitialsColor.green,
            data: '25%',
        },
        {
            imageUrl: testImages_1.TestImages.personaFemale,
            imageInitials: 'VL',
            personaName: 'Valentina Lovric',
            initialsColor: PersonaInitialsColor.lightBlue,
            data: 'Emp1234',
            onClick: function (ev, persona) {
                return alert('You clicked on ' + persona.personaName + '. Extra data: ' + persona.data);
            },
        },
        {
            imageUrl: testImages_1.TestImages.personaMale,
            imageInitials: 'MS',
            personaName: 'Maor Sharett',
            initialsColor: PersonaInitialsColor.lightGreen,
        },
        {
            imageUrl: testImages_1.TestImages.personaFemale,
            imageInitials: 'PV',
            personaName: 'Annie Lindqvist2',
            initialsColor: PersonaInitialsColor.lightPink,
        },
        {
            imageUrl: testImages_1.TestImages.personaMale,
            imageInitials: 'AR',
            personaName: 'Aaron Reid2',
            initialsColor: PersonaInitialsColor.magenta,
            data: 'Emp1234',
            onClick: function (ev, persona) {
                return alert('You clicked on ' + persona.personaName + '. Extra data: ' + persona.data);
            },
        },
        {
            imageUrl: testImages_1.TestImages.personaMale,
            imageInitials: 'AL',
            personaName: 'Alex Lundberg2',
            initialsColor: PersonaInitialsColor.orange,
        },
        {
            imageUrl: testImages_1.TestImages.personaMale,
            imageInitials: 'RK',
            personaName: 'Roko Kolar2',
            initialsColor: PersonaInitialsColor.pink,
        },
        {
            imageUrl: testImages_1.TestImages.personaMale,
            imageInitials: 'CB',
            personaName: 'Christian Bergqvist2',
            initialsColor: PersonaInitialsColor.purple,
        },
        {
            imageUrl: testImages_1.TestImages.personaFemale,
            imageInitials: 'VL',
            personaName: 'Valentina Lovric2',
            initialsColor: PersonaInitialsColor.red,
        },
        {
            imageUrl: testImages_1.TestImages.personaMale,
            imageInitials: 'MS',
            personaName: 'Maor Sharett2',
            initialsColor: PersonaInitialsColor.teal,
        },
        {
            imageUrl: testImages_1.TestImages.personaFemale,
            imageInitials: 'VL',
            personaName: 'Another A Name',
            initialsColor: PersonaInitialsColor.blue,
        },
        {
            imageUrl: testImages_1.TestImages.personaMale,
            imageInitials: 'MS',
            personaName: 'Another A Name (So Many A names!)',
            initialsColor: PersonaInitialsColor.darkBlue,
        },
        {
            imageUrl: testImages_1.TestImages.personaFemale,
            imageInitials: 'VL',
            personaName: 'Another Anecdotal A Name',
            initialsColor: PersonaInitialsColor.darkGreen,
        },
        {
            imageUrl: testImages_1.TestImages.personaMale,
            imageInitials: 'MS',
            personaName: 'Anerobic A Name',
            initialsColor: PersonaInitialsColor.darkRed,
        },
        {
            imageUrl: testImages_1.TestImages.personaFemale,
            imageInitials: 'VL',
            personaName: 'Aerobic A Name',
            initialsColor: PersonaInitialsColor.green,
        },
        {
            imageUrl: testImages_1.TestImages.personaMale,
            imageInitials: 'MS',
            personaName: 'Maor Sharett2',
            initialsColor: PersonaInitialsColor.lightBlue,
        },
        {
            imageUrl: testImages_1.TestImages.personaFemale,
            imageInitials: 'VL',
            personaName: 'Valentina Lovric2',
            initialsColor: PersonaInitialsColor.lightGreen,
        },
        {
            imageUrl: testImages_1.TestImages.personaMale,
            imageInitials: 'MS',
            personaName: 'Maor Sharett2',
            initialsColor: PersonaInitialsColor.lightPink,
        },
        {
            imageUrl: testImages_1.TestImages.personaFemale,
            imageInitials: 'VL',
            personaName: 'Valentina Lovric2',
            initialsColor: PersonaInitialsColor.magenta,
        },
        {
            imageUrl: testImages_1.TestImages.personaMale,
            imageInitials: 'MS',
            personaName: 'Maor Sharett2',
            initialsColor: PersonaInitialsColor.orange,
        },
    ];
});
