// @fluentui/react-dialog/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Dialog: ()=>_dialog.Dialog,
    renderDialog_unstable: ()=>_dialog.renderDialog_unstable,
    useDialog_unstable: ()=>_dialog.useDialog_unstable,
    DialogTrigger: ()=>_dialogTrigger.DialogTrigger,
    useDialogTrigger_unstable: ()=>_dialogTrigger.useDialogTrigger_unstable,
    renderDialogTrigger_unstable: ()=>_dialogTrigger.renderDialogTrigger_unstable,
    DialogActions: ()=>_dialogActions.DialogActions,
    dialogActionsClassNames: ()=>_dialogActions.dialogActionsClassNames,
    useDialogActions_unstable: ()=>_dialogActions.useDialogActions_unstable,
    useDialogActionsStyles_unstable: ()=>_dialogActions.useDialogActionsStyles_unstable,
    renderDialogActions_unstable: ()=>_dialogActions.renderDialogActions_unstable,
    DialogBody: ()=>_dialogBody.DialogBody,
    dialogBodyClassNames: ()=>_dialogBody.dialogBodyClassNames,
    useDialogBody_unstable: ()=>_dialogBody.useDialogBody_unstable,
    useDialogBodyStyles_unstable: ()=>_dialogBody.useDialogBodyStyles_unstable,
    renderDialogBody_unstable: ()=>_dialogBody.renderDialogBody_unstable,
    DialogTitle: ()=>_dialogTitle.DialogTitle,
    dialogTitleClassNames: ()=>_dialogTitle.dialogTitleClassNames,
    useDialogTitle_unstable: ()=>_dialogTitle.useDialogTitle_unstable,
    useDialogTitleStyles_unstable: ()=>_dialogTitle.useDialogTitleStyles_unstable,
    renderDialogTitle_unstable: ()=>_dialogTitle.renderDialogTitle_unstable,
    DialogSurface: ()=>_dialogSurface.DialogSurface,
    dialogSurfaceClassNames: ()=>_dialogSurface.dialogSurfaceClassNames,
    useDialogSurface_unstable: ()=>_dialogSurface.useDialogSurface_unstable,
    useDialogSurfaceStyles_unstable: ()=>_dialogSurface.useDialogSurfaceStyles_unstable,
    renderDialogSurface_unstable: ()=>_dialogSurface.renderDialogSurface_unstable,
    DialogContent: ()=>_dialogContent.DialogContent,
    dialogContentClassNames: ()=>_dialogContent.dialogContentClassNames,
    useDialogContent_unstable: ()=>_dialogContent.useDialogContent_unstable,
    useDialogContentStyles_unstable: ()=>_dialogContent.useDialogContentStyles_unstable,
    renderDialogContent_unstable: ()=>_dialogContent.renderDialogContent_unstable
});
const _dialog = require("./Dialog");
const _dialogTrigger = require("./DialogTrigger");
const _dialogActions = require("./DialogActions");
const _dialogBody = require("./DialogBody");
const _dialogTitle = require("./DialogTitle");
const _dialogSurface = require("./DialogSurface");
const _dialogContent = require("./DialogContent");
