// @fluentui/react-radio/lib-commonjs/components/RadioGroup/useRadioGroup.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useRadioGroup_unstable", {
    enumerable: true,
    get: ()=>useRadioGroup_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactUtilities = require("@fluentui/react-utilities");
const useRadioGroup_unstable = (props, ref)=>{
    const generatedName = (0, _reactUtilities.useId)('radiogroup-');
    const { name =generatedName , value , defaultValue , disabled , layout ='vertical' , onChange , required  } = props;
    return {
        layout,
        name,
        value,
        defaultValue,
        disabled,
        required,
        components: {
            root: 'div'
        },
        root: {
            ref,
            role: 'radiogroup',
            ...(0, _reactUtilities.getNativeElementProps)('div', props, /*excludedPropNames:*/ [
                'onChange',
                'name'
            ]),
            onChange: (0, _reactUtilities.useEventCallback)((ev)=>{
                if (onChange && (0, _reactUtilities.isHTMLElement)(ev.target, {
                    constructorName: 'HTMLInputElement'
                }) && ev.target.type === 'radio') {
                    onChange(ev, {
                        value: ev.target.value
                    });
                }
            })
        }
    };
}; //# sourceMappingURL=useRadioGroup.js.map
