// @fluentui/react-text/lib-commonjs/components/presets/Title3/Title3.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    title3ClassNames: ()=>title3ClassNames,
    Title3: ()=>Title3
});
const _react = require("@griffel/react");
const _createPreset = require("../createPreset");
const title3ClassNames = {
    root: 'fui-Title3'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "f1x0m3f5",
        Bhrd7zp: "fl43uef",
        Bg96gwp: "fb86gi6"
    }
}, {
    d: [
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".f1x0m3f5{font-size:var(--fontSizeBase600);}",
        ".fl43uef{font-weight:var(--fontWeightSemibold);}",
        ".fb86gi6{line-height:var(--lineHeightBase600);}"
    ]
});
const Title3 = /*#__PURE__*/ (0, _createPreset.createPreset)({
    useStyles,
    className: title3ClassNames.root,
    displayName: 'Title3'
}); //# sourceMappingURL=Title3.js.map
