// @fluentui/react-text/lib-commonjs/components/presets/Caption2/Caption2.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    caption2ClassNames: ()=>caption2ClassNames,
    Caption2: ()=>Caption2
});
const _react = require("@griffel/react");
const _createPreset = require("../createPreset");
const caption2ClassNames = {
    root: 'fui-Caption2'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "f13mqy1h",
        Bhrd7zp: "figsok6",
        Bg96gwp: "fcpl73t"
    }
}, {
    d: [
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".f13mqy1h{font-size:var(--fontSizeBase100);}",
        ".figsok6{font-weight:var(--fontWeightRegular);}",
        ".fcpl73t{line-height:var(--lineHeightBase100);}"
    ]
});
const Caption2 = /*#__PURE__*/ (0, _createPreset.createPreset)({
    useStyles,
    className: caption2ClassNames.root,
    displayName: 'Caption2'
}); //# sourceMappingURL=Caption2.js.map
