// @fluentui/react-menu/lib-commonjs/contexts/menuListContext.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    MenuListContext: ()=>MenuListContext,
    MenuListProvider: ()=>MenuListProvider,
    useMenuListContext_unstable: ()=>useMenuListContext_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactContextSelector = require("@fluentui/react-context-selector");
const MenuListContext = /*#__PURE__*/ (0, _reactContextSelector.createContext)(undefined);
const menuListContextDefaultValue = {
    checkedValues: {},
    setFocusByFirstCharacter: ()=>null,
    toggleCheckbox: ()=>null,
    selectRadio: ()=>null,
    hasIcons: false,
    hasCheckmarks: false
};
const MenuListProvider = MenuListContext.Provider;
const useMenuListContext_unstable = (selector)=>(0, _reactContextSelector.useContextSelector)(MenuListContext, (ctx = menuListContextDefaultValue)=>selector(ctx)); //# sourceMappingURL=menuListContext.js.map
