// @fluentui/react-text/lib-commonjs/components/presets/Body2/Body2.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    body2ClassNames: ()=>body2ClassNames,
    Body2: ()=>Body2
});
const _react = require("@griffel/react");
const _createPreset = require("../createPreset");
const body2ClassNames = {
    root: 'fui-Body2'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "fod5ikn",
        Bhrd7zp: "figsok6",
        Bg96gwp: "faaz57k"
    }
}, {
    d: [
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".fod5ikn{font-size:var(--fontSizeBase400);}",
        ".figsok6{font-weight:var(--fontWeightRegular);}",
        ".faaz57k{line-height:var(--lineHeightBase400);}"
    ]
});
const Body2 = /*#__PURE__*/ (0, _createPreset.createPreset)({
    useStyles,
    className: body2ClassNames.root,
    displayName: 'Body2'
}); //# sourceMappingURL=Body2.js.map
