// @fluentui/react-infobutton/lib-commonjs/components/InfoLabel/useInfoLabel.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useInfoLabel_unstable", {
    enumerable: true,
    get: ()=>useInfoLabel_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactLabel = require("@fluentui/react-label");
const _reactUtilities = require("@fluentui/react-utilities");
const _infoButton = require("../InfoButton/InfoButton");
var _infoButton1, _arialabelledby;
const useInfoLabel_unstable = (props, ref)=>{
    const { root: rootShorthand , label: labelShorthand , infoButton: infoButtonShorthand , size , info , className , style , ...labelProps } = props;
    const root = (0, _reactUtilities.resolveShorthand)(rootShorthand, {
        required: true,
        defaultProps: {
            className,
            style
        }
    });
    const label = (0, _reactUtilities.resolveShorthand)(labelShorthand, {
        required: true,
        defaultProps: {
            id: (0, _reactUtilities.useId)('infolabel-'),
            ref,
            size,
            ...labelProps
        }
    });
    const infoButton = (0, _reactUtilities.resolveShorthand)(infoButtonShorthand, {
        required: !!info,
        defaultProps: {
            id: (0, _reactUtilities.useId)('infobutton-'),
            content: info,
            size
        }
    });
    if (infoButton) {
        var _;
        (_ = (_infoButton1 = infoButton)[_arialabelledby = 'aria-labelledby']) !== null && _ !== void 0 ? _ : _infoButton1[_arialabelledby] = `${label.id} ${infoButton.id}`;
    }
    return {
        size,
        components: {
            root: 'span',
            label: _reactLabel.Label,
            infoButton: _infoButton.InfoButton
        },
        root,
        label,
        infoButton
    };
}; //# sourceMappingURL=useInfoLabel.js.map
