// @fluentui/react-utilities/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    getSlots: ()=>_index.getSlots,
    resolveShorthand: ()=>_index.resolveShorthand,
    isResolvedShorthand: ()=>_index.isResolvedShorthand,
    IdPrefixProvider: ()=>_index1.IdPrefixProvider,
    resetIdsForTests: ()=>_index1.resetIdsForTests,
    useControllableState: ()=>_index1.useControllableState,
    useEventCallback: ()=>_index1.useEventCallback,
    useFirstMount: ()=>_index1.useFirstMount,
    useForceUpdate: ()=>_index1.useForceUpdate,
    useId: ()=>_index1.useId,
    useIsomorphicLayoutEffect: ()=>_index1.useIsomorphicLayoutEffect,
    useMergedRefs: ()=>_index1.useMergedRefs,
    useOnClickOutside: ()=>_index1.useOnClickOutside,
    useOnScrollOutside: ()=>_index1.useOnScrollOutside,
    usePrevious: ()=>_index1.usePrevious,
    useScrollbarWidth: ()=>_index1.useScrollbarWidth,
    useTimeout: ()=>_index1.useTimeout,
    canUseDOM: ()=>_index2.canUseDOM,
    useIsSSR: ()=>_index2.useIsSSR,
    SSRProvider: ()=>_index2.SSRProvider,
    clamp: ()=>_index3.clamp,
    getNativeElementProps: ()=>_index3.getNativeElementProps,
    getPartitionedNativeProps: ()=>_index3.getPartitionedNativeProps,
    getRTLSafeKey: ()=>_index3.getRTLSafeKey,
    mergeCallbacks: ()=>_index3.mergeCallbacks,
    isHTMLElement: ()=>_index3.isHTMLElement,
    isInteractiveHTMLElement: ()=>_index3.isInteractiveHTMLElement,
    omit: ()=>_index3.omit,
    applyTriggerPropsToChildren: ()=>_index4.applyTriggerPropsToChildren,
    getTriggerChild: ()=>_index4.getTriggerChild,
    isFluentTrigger: ()=>_index4.isFluentTrigger,
    isTouchEvent: ()=>_index5.isTouchEvent,
    isMouseEvent: ()=>_index5.isMouseEvent,
    getEventClientCoords: ()=>_index5.getEventClientCoords
});
const _index = require("./compose/index");
const _index1 = require("./hooks/index");
const _index2 = require("./ssr/index");
const _index3 = require("./utils/index");
const _index4 = require("./trigger/index");
const _index5 = require("./events/index");
