// @fluentui/react-button/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Button: ()=>_button.Button,
    buttonClassNames: ()=>_button.buttonClassNames,
    renderButton_unstable: ()=>_button.renderButton_unstable,
    useButtonStyles_unstable: ()=>_button.useButtonStyles_unstable,
    useButton_unstable: ()=>_button.useButton_unstable,
    CompoundButton: ()=>_compoundButton.CompoundButton,
    compoundButtonClassNames: ()=>_compoundButton.compoundButtonClassNames,
    renderCompoundButton_unstable: ()=>_compoundButton.renderCompoundButton_unstable,
    useCompoundButtonStyles_unstable: ()=>_compoundButton.useCompoundButtonStyles_unstable,
    useCompoundButton_unstable: ()=>_compoundButton.useCompoundButton_unstable,
    MenuButton: ()=>_menuButton.MenuButton,
    menuButtonClassNames: ()=>_menuButton.menuButtonClassNames,
    renderMenuButton_unstable: ()=>_menuButton.renderMenuButton_unstable,
    useMenuButtonStyles_unstable: ()=>_menuButton.useMenuButtonStyles_unstable,
    useMenuButton_unstable: ()=>_menuButton.useMenuButton_unstable,
    SplitButton: ()=>_splitButton.SplitButton,
    renderSplitButton_unstable: ()=>_splitButton.renderSplitButton_unstable,
    splitButtonClassNames: ()=>_splitButton.splitButtonClassNames,
    useSplitButtonStyles_unstable: ()=>_splitButton.useSplitButtonStyles_unstable,
    useSplitButton_unstable: ()=>_splitButton.useSplitButton_unstable,
    ToggleButton: ()=>_toggleButton.ToggleButton,
    renderToggleButton_unstable: ()=>_toggleButton.renderToggleButton_unstable,
    toggleButtonClassNames: ()=>_toggleButton.toggleButtonClassNames,
    useToggleButtonStyles_unstable: ()=>_toggleButton.useToggleButtonStyles_unstable,
    useToggleButton_unstable: ()=>_toggleButton.useToggleButton_unstable,
    useToggleState: ()=>_index.useToggleState,
    ButtonContextProvider: ()=>_index1.ButtonContextProvider,
    useButtonContext: ()=>_index1.useButtonContext
});
const _button = require("./Button");
const _compoundButton = require("./CompoundButton");
const _menuButton = require("./MenuButton");
const _splitButton = require("./SplitButton");
const _toggleButton = require("./ToggleButton");
const _index = require("./utils/index");
const _index1 = require("./contexts/index");
