// @fluentui/react-text/lib-commonjs/components/presets/Caption1Strong/Caption1Strong.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    caption1StrongClassNames: ()=>caption1StrongClassNames,
    Caption1Strong: ()=>Caption1Strong
});
const _react = require("@griffel/react");
const _createPreset = require("../createPreset");
const caption1StrongClassNames = {
    root: 'fui-Caption1Strong'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "fy9rknc",
        Bhrd7zp: "fl43uef",
        Bg96gwp: "fwrc4pm"
    }
}, {
    d: [
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".fy9rknc{font-size:var(--fontSizeBase200);}",
        ".fl43uef{font-weight:var(--fontWeightSemibold);}",
        ".fwrc4pm{line-height:var(--lineHeightBase200);}"
    ]
});
const Caption1Strong = /*#__PURE__*/ (0, _createPreset.createPreset)({
    useStyles,
    className: caption1StrongClassNames.root,
    displayName: 'Caption1Strong'
}); //# sourceMappingURL=Caption1Strong.js.map
