// @fluentui/react-popover/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Popover: ()=>_popover.Popover,
    renderPopover_unstable: ()=>_popover.renderPopover_unstable,
    usePopover_unstable: ()=>_popover.usePopover_unstable,
    PopoverSurface: ()=>_popoverSurface.PopoverSurface,
    arrowHeights: ()=>_popoverSurface.arrowHeights,
    popoverSurfaceClassNames: ()=>_popoverSurface.popoverSurfaceClassNames,
    renderPopoverSurface_unstable: ()=>_popoverSurface.renderPopoverSurface_unstable,
    usePopoverSurfaceStyles_unstable: ()=>_popoverSurface.usePopoverSurfaceStyles_unstable,
    usePopoverSurface_unstable: ()=>_popoverSurface.usePopoverSurface_unstable,
    PopoverProvider: ()=>_popoverContext.PopoverProvider,
    usePopoverContext_unstable: ()=>_popoverContext.usePopoverContext_unstable,
    PopoverTrigger: ()=>_popoverTrigger.PopoverTrigger,
    renderPopoverTrigger_unstable: ()=>_popoverTrigger.renderPopoverTrigger_unstable,
    usePopoverTrigger_unstable: ()=>_popoverTrigger.usePopoverTrigger_unstable
});
const _popover = require("./Popover");
const _popoverSurface = require("./PopoverSurface");
const _popoverContext = require("./popoverContext");
const _popoverTrigger = require("./PopoverTrigger");
