// @fluentui/react-text/lib-commonjs/components/presets/Caption2Strong/Caption2Strong.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    caption2StrongClassNames: ()=>caption2StrongClassNames,
    Caption2Strong: ()=>Caption2Strong
});
const _react = require("@griffel/react");
const _createPreset = require("../createPreset");
const caption2StrongClassNames = {
    root: 'fui-Caption2Strong'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "f13mqy1h",
        Bhrd7zp: "fl43uef",
        Bg96gwp: "fcpl73t"
    }
}, {
    d: [
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".f13mqy1h{font-size:var(--fontSizeBase100);}",
        ".fl43uef{font-weight:var(--fontWeightSemibold);}",
        ".fcpl73t{line-height:var(--lineHeightBase100);}"
    ]
});
const Caption2Strong = /*#__PURE__*/ (0, _createPreset.createPreset)({
    useStyles,
    className: caption2StrongClassNames.root,
    displayName: 'Caption2Strong'
}); //# sourceMappingURL=Caption2Strong.js.map
