// @fluentui/react-dialog/lib-commonjs/contexts/dialogContext.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    DialogContext: ()=>DialogContext,
    DialogProvider: ()=>DialogProvider,
    useDialogContext_unstable: ()=>useDialogContext_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactContextSelector = require("@fluentui/react-context-selector");
const defaultContextValue = {
    open: false,
    inertTrapFocus: false,
    modalType: 'modal',
    isNestedDialog: false,
    dialogRef: {
        current: null
    },
    requestOpenChange () {
    /* noop */ }
};
const DialogContext = /*#__PURE__*/ (0, _reactContextSelector.createContext)(undefined);
const DialogProvider = DialogContext.Provider;
const useDialogContext_unstable = (selector)=>(0, _reactContextSelector.useContextSelector)(DialogContext, (ctx = defaultContextValue)=>selector(ctx)); //# sourceMappingURL=dialogContext.js.map
