// @fluentui/react-text/lib-commonjs/components/presets/Caption1Stronger/Caption1Stronger.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    caption1StrongerClassNames: ()=>caption1StrongerClassNames,
    Caption1Stronger: ()=>Caption1Stronger
});
const _react = require("@griffel/react");
const _createPreset = require("../createPreset");
const caption1StrongerClassNames = {
    root: 'fui-Caption1Stronger'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "fy9rknc",
        Bhrd7zp: "flh3ekv",
        Bg96gwp: "fwrc4pm"
    }
}, {
    d: [
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".fy9rknc{font-size:var(--fontSizeBase200);}",
        ".flh3ekv{font-weight:var(--fontWeightBold);}",
        ".fwrc4pm{line-height:var(--lineHeightBase200);}"
    ]
});
const Caption1Stronger = /*#__PURE__*/ (0, _createPreset.createPreset)({
    useStyles,
    className: caption1StrongerClassNames.root,
    displayName: 'Caption1Stronger'
}); //# sourceMappingURL=Caption1Stronger.js.map
