// @fluentui/react-virtualizer/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Virtualizer: ()=>_virtualizer.Virtualizer,
    virtualizerClassNames: ()=>_virtualizer.virtualizerClassNames,
    useVirtualizer_unstable: ()=>_virtualizer.useVirtualizer_unstable,
    renderVirtualizer_unstable: ()=>_virtualizer.renderVirtualizer_unstable,
    useVirtualizerStyles_unstable: ()=>_virtualizer.useVirtualizerStyles_unstable,
    useIntersectionObserver: ()=>_hooks.useIntersectionObserver,
    useStaticVirtualizerMeasure: ()=>_hooks.useStaticVirtualizerMeasure,
    VirtualizerScrollView: ()=>_virtualizerScrollView.VirtualizerScrollView,
    virtualizerScrollViewClassNames: ()=>_virtualizerScrollView.virtualizerScrollViewClassNames,
    useVirtualizerScrollView_unstable: ()=>_virtualizerScrollView.useVirtualizerScrollView_unstable,
    renderVirtualizerScrollView_unstable: ()=>_virtualizerScrollView.renderVirtualizerScrollView_unstable,
    useVirtualizerScrollViewStyles_unstable: ()=>_virtualizerScrollView.useVirtualizerScrollViewStyles_unstable
});
const _virtualizer = require("./Virtualizer");
const _hooks = require("./Hooks");
const _virtualizerScrollView = require("./VirtualizerScrollView");
